src/utilities/scroll-utils.test.ts

```diff
@@ -36,6 +36,37 @@ describe("ScrollUtils", () => {
             // Assert
             expect(getElementByIdMock).toBeCalledTimes(50);
         });
+
+        test("when element is not found after 50 attempts, it logs a console warning for development environment", () => {
```

Comment:
@SaidShah as an outsider reading this test (not knowing the implementation), I don't see where the '50 attempts' statement is tested? Is that something internal the implementation? If so, be aware that tests should try to be an "outsider" like I am. It shouldn't do things assuming internals of the tested party it has no control over. 

To reconcile the above, you might just simplify the test working by removing `after 50 attempts`?